File: server/dto/match/match_context_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.misc.cell_dto import CellDto
from dto.player.player_dto import PlayerDto


@dataclass
class MatchContextDto(BaseDto):
    id: str
    roomId: str
    boardArray: list[list[CellDto]]
    currentTurn: int
    player1: PlayerDto
    player2: PlayerDto

File: client/src/dto/match/MatchContextDto.ts
import { EMPTY_STRING } from "../../env";
import { CellDto } from "../cell/CellDto";
import { PlayerResourceBundleDto, undefinedPlayerResourceBundleDto } from "../player/PlayerInfoBundleDto";

export interface MatchContextDto {
    id: string;
    roomId: string;
    boardArray: CellDto[][];
    currentTurn: number;
    playerResourcesBundle: PlayerResourceBundleDto;
}

export const undefinedMatch: MatchContextDto = {
    id: EMPTY_STRING,
    roomId: EMPTY_STRING,
    boardArray: [],
    currentTurn: 0,
    playerResourcesBundle: undefinedPlayerResourceBundleDto
};